projects/js-packages/publicize-components/src/components/social-post-modal/modal.tsx

```diff
@@ -15,7 +16,19 @@ import styles from './styles.module.scss';
  * @return {import('react').ReactNode} - Social Post Modal component.
  */
 export function SocialPostModal() {
-	const [ isModalOpen, toggleModal ] = useReducer( state => ! state, false );
+	const [ isModalOpen, toggleModal ] = useReducer(
+		state => {
+			const url = new URL( window.location.href );
+			// If the modal is opened with the share post query arg, remove it from the URL.
+			if ( state && url.searchParams.has( 'jetpackSidebar' ) ) {
+				url.searchParams.delete( 'jetpackSidebar' );
+				window.history.replaceState( null, '', url.toString() );
+			}
+			return ! state;
+		},
+		null,
+		() => getQueryArg( window.location.search, 'jetpackSidebar' ) === 'open_with_share_post'
```

Comment:
I wonder if this should be here also. I think it might be better to do all the query parameter checking in one place, and perhaps we could pass in the initial state of the modal?

Reply:
Done in 3dd0fe7326db1d4a0886ed2d69b017b4cb5a6cbe

Thanks

Reply:
Ah OK, that means we have to make that modal state global.